### ISSUE TITLE
Improve code sample.

### ISSUE THREAD

**Issue by filipedelimabrito** [google/material-design-lite#4697]
As the creation of checkbox is being deprecated and we are now responsible for creating and handling them, I think that is a good idea improve at least the code sample on [wiki page](https://github.com/google/material-design-lite/wiki/Deprecations). 

When implementing the code sample I am not able to see the table rows selected. Then, to see the rows selected, I did it:

```javascript
var table = document.querySelector('table');
var headerCheckbox = table.querySelector('thead .mdl-data-table__select input');
var boxes = table.querySelectorAll('tbody .mdl-data-table__select');
var rows = table.querySelectorAll('tbody tr');

var headerCheckHandler = function(event) {
  if (event.target.checked) {
    for (var i = 0, length = boxes.length; i < length; i++) {
        boxes[i].MaterialCheckbox.check();
        rows[i].classList.add('is-selected');
    }
  }else {
    for (var i = 0, length = boxes.length; i < length; i++) {
        boxes[i].MaterialCheckbox.uncheck();
        rows[i].classList.remove('is-selected');
    }
  }
};
headerCheckbox.addEventListener('change', headerCheckHandler);
```

There other points to consider (for example: Select a row when a user click on the row's checkbox and uncheck the header's checkbox when some row's checkbox is unchecked), but it is a start point to me.

Thank MDL developer team (especially to @Garbee who provided us the code example).

**Comment by Garbee** [comment]
We aren't going to go into any more detail with the wiki example. If people want that functionality, they can peek at the source we had used as a starting point.

The whole reason for deprecating that is because it was confusing and there are too many edge-cases going on. Instead of trying to handle all the cases and continue providing an inferior user experience by supporting that component, we dumped it.

The example shouldn't provide everything developers may need. At that point it isn't an example for getting the same interface. Some work is required here and putting more effort into the documentation samples for that isn't going to end up providing a good developer experience.

Overall, the checking/unchecking logic is pretty basic JavaScript (aside from needing to know our component calls.) I don't feel there is enough return on going through deeper documentation when we are focusing on V2 at this point. Especially when under the "Considerations" section it makes it clear developers are responsible for some logic on their own.

I did go add the call to `updateClasses` manually to make sure the classes get updated (check/uncheck should do it already iirc.) Other than this, no action will be taken to improve the sample code, as it is only a skeleton of logic and not intended as an absolute replacement to what was provided before.

Thank you.
